docs/docsite/rst/user_guide/playbooks_reuse_roles.rst

```diff
@@ -256,12 +258,14 @@ You can pass other keywords, including variables and tags, when importing roles:
 
 When you add a tag to an ``import_role`` statement, Ansible applies the tag to `all` tasks within the role. See :ref:`tag_inheritance` for details.
 
+Variables of the role (from section `vars:` and from `roles/x/vars/main.yml`) are available to all tasks and all roles of the play.
+
 .. _run_role_twice:
 
 Running a role multiple times in one playbook
 =============================================
 
-Ansible only executes each role once, even if you define it multiple times, unless the parameters defined on the role are different for each definition. For example, Ansible only runs the role ``foo`` once in a play like this:
+Ansible only executes each role once, even if you define it multiple times, unless the variabless defined on the role are different for each definition. For example, Ansible only runs the role ``foo`` once in a play like this:
```

Comment:
variabless -> variables

Reply:
>  variabless -> variables

True, I will fix this.

Reply:
This line is what I understand how it works. But the current devel document writes something different:

https://github.com/ansible/ansible/blob/05662fd/docs/docsite/rst/user_guide/playbooks_reuse_roles.rst#passing-different-parameters writes that you have to use parameters. My experiments with some older version showed that this is not totally true. But you must carefully define all variables in each invocation (because of variable scoping, which is buggy in my opinion #50278).

https://github.com/ansible/ansible/blob/05662fd/docs/docsite/rst/user_guide/playbooks_reuse_roles.rst#running-role-dependencies-multiple-times-in-one-playbook mentions that variable scoping changes.